layered-architectures-module-2/Carting.DAL/Repositories/IMongoRepository.cs

```diff
@@ -0,0 +1,11 @@
+﻿using Carting.DAL.Models;
+
+namespace Carting.DAL.Repositories;
+
+public interface IMongoRepository<T> where T:EntityBase
+{
+    Task GetByIdAsync(string id);
+    Task<IEnumerable<T>> GetAllAsync();
+    Task CreateAsync(T obj);
+    Task UpdateAsync(T obj);
```

Comment:
what about delete?

Reply:
it's not required for the task but you must have this method to complete repository pattern

Reply:
Added